workspace/pkgs-graph/src/index.ts

```diff
@@ -35,7 +35,14 @@ export function createPkgGraph<T> (pkgs: Array<Package & T>, opts?: {
     unmatched: Array<{ pkgName: string, range: string }>
   } {
   const pkgMap = createPkgMap(pkgs)
-  const pkgMapValues = Object.values(pkgMap)
+  const pkgMapByManifestName: Record<string, Package[] | undefined> = {}
+  for (const key in pkgMap) {
+    const pkg = pkgMap[key]
```

Comment:
Usually when I see

```ts
for (const key in obj) {
  const value = obj[key]
  // ...
}
```

My brain immediately thinks that can be refactored to:

```ts
for (const value of Object.values(obj)) {
  // ...
}
```

It took me a minute to realize we're intentionally avoiding the more concise pattern. Can we add a comment that we're avoiding `Object.values(pkgMap)` here for performance reasons? The comment makes it less likely someone in the future will regress this.

Reply:
This code only executes once, so it's possible that there's no need to actually optimize it as much as I have; I'll retest this with the more obvious code instead tomorrow. I bet the runtime can optimize it, or at least it won't actually be any different.

Reply:
Thanks! For what it's worth, I do think the `pkgMapByManifestName` map is likely where most of the performance gains posted came from. Saving a `Object.values(...)` call may decrease memory usage if V8 doesn't optimize that, but I'd be surprised if CPU profiling showed that saved a non-negligible amount of time.

Reply:
Good news, it's not slower to do this. Also means I can drop the lazy code from below instead because I already need `Object.values(pkgMap)` for this loop.